vehicle/as/include/ssc_interface/ssc_interface.h

```diff
@@ -17,89 +17,99 @@
+
+  std::unique_ptr<message_filters::Subscriber<automotive_platform_msgs::msg::VelocityAccelCov>>
+    velocity_accel_cov_sub_;
+  std::unique_ptr<message_filters::Subscriber<automotive_platform_msgs::msg::CurvatureFeedback>>
     curvature_feedback_sub_;
-  message_filters::Subscriber<automotive_platform_msgs::ThrottleFeedback> * throttle_feedback_sub_;
-  message_filters::Subscriber<automotive_platform_msgs::BrakeFeedback> * brake_feedback_sub_;
-  message_filters::Subscriber<automotive_platform_msgs::GearFeedback> * gear_feedback_sub_;
-  message_filters::Subscriber<pacmod_msgs::WheelSpeedRpt> * wheel_speed_sub_;
-  message_filters::Subscriber<pacmod_msgs::SystemRptFloat> * steering_wheel_sub_;
-  message_filters::Synchronizer<SSCFeedbacksSyncPolicy> * ssc_feedbacks_sync_;
-
-  ros::Subscriber pacmod_turn_sub_;  // TEMP to support turn_signal
+  std::unique_ptr<message_filters::Subscriber<automotive_platform_msgs::msg::ThrottleFeedback>>
+    throttle_feedback_sub_;
+  std::unique_ptr<message_filters::Subscriber<automotive_platform_msgs::msg::BrakeFeedback>>
+    brake_feedback_sub_;
+  std::unique_ptr<message_filters::Subscriber<automotive_platform_msgs::msg::GearFeedback>>
+    gear_feedback_sub_;
+  std::unique_ptr<message_filters::Subscriber<pacmod_msgs::msg::WheelSpeedRpt>> wheel_speed_sub_;
+  std::unique_ptr<message_filters::Subscriber<pacmod_msgs::msg::SystemRptFloat>>
+    steering_wheel_sub_;
+  std::unique_ptr<message_filters::Synchronizer<SSCFeedbacksSyncPolicy>> ssc_feedbacks_sync_;
+
+  // TEMP to support turn_signal
+  rclcpp::Subscription<pacmod_msgs::msg::SystemRptInt>::SharedPtr pacmod_turn_sub_;
 
   // publishers
-  ros::Publisher steer_mode_pub_;
-  ros::Publisher speed_mode_pub_;
-  ros::Publisher turn_signal_pub_;
-  ros::Publisher gear_pub_;
-  ros::Publisher current_shift_pub_;
-  ros::Publisher control_mode_pub_;
-  ros::Publisher current_twist_pub_;
-  ros::Publisher current_steer_pub_;
-  ros::Publisher current_steer_wheel_deg_pub_;
-  ros::Publisher current_velocity_pub_;
-  ros::Publisher current_velocity_kmph_pub_;
-  ros::Publisher current_turn_signal_pub_;
+  rclcpp::Publisher<automotive_platform_msgs::msg::SteerMode>::SharedPtr steer_mode_pub_;
+  rclcpp::Publisher<automotive_platform_msgs::msg::SpeedMode>::SharedPtr speed_mode_pub_;
+  rclcpp::Publisher<automotive_platform_msgs::msg::TurnSignalCommand>::SharedPtr turn_signal_pub_;
+  rclcpp::Publisher<automotive_platform_msgs::msg::GearCommand>::SharedPtr gear_pub_;
+  rclcpp::Publisher<autoware_vehicle_msgs::msg::ShiftStamped>::SharedPtr current_shift_pub_;
+  rclcpp::Publisher<autoware_vehicle_msgs::msg::ControlMode>::SharedPtr control_mode_pub_;
+  rclcpp::Publisher<geometry_msgs::msg::TwistStamped>::SharedPtr current_twist_pub_;
+  rclcpp::Publisher<autoware_vehicle_msgs::msg::Steering>::SharedPtr current_steer_pub_;
+  rclcpp::Publisher<std_msgs::msg::Float32>::SharedPtr current_steer_wheel_deg_pub_;
```

Comment:
Consider creating wrapper types for Float32 messages etc. since they are deprecated in Foxy